services/submission/src/main/java/app/coronawarn/server/services/submission/controller/SubmissionController.java

```diff
@@ -126,12 +128,13 @@
     List<DiagnosisKey> diagnosisKeys = new ArrayList<>();
 
     for (TemporaryExposureKey protoBufferKey : protoBufferKeys) {
-      String originCountry = StringUtils.defaultIfBlank(submissionPayload.getOrigin(),
-          submissionServiceConfig.getDefaultOriginCountry());
+      String originCountry = defaultIfEmptyOriginCountry(submissionPayload.getOrigin());
+      List<String> visitedCountries = extendVisitedCountriesWithOriginCountry(
+          submissionPayload.getVisitedCountriesList());
 
       DiagnosisKey diagnosisKey = DiagnosisKey.builder()
           .fromTemporaryExposureKey(protoBufferKey)
-          .withVisitedCountries(submissionPayload.getVisitedCountriesList())
+          .withVisitedCountries(visitedCountries)
           .withCountryCode(originCountry)
           .withConsentToFederation(submissionPayload.getConsentToFederation())
           .withFieldNormalization(new SubmissionKeyNormalizer(submissionServiceConfig))
```

Comment:
Did you consider extracting this to the DiagnosisKeyBuilder? Maybe it makes sense to concentrate the builder logic there for later reuse

Reply:
Might be difficult since it's part of another module where we don't have the origin country parameter